tools/c7n_azure/tests_azure/test_resource_graph_source.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from datetime import timedelta
+
+from tests_azure.azure_common import BaseTest, arm_template
+from dateutil.parser import parse
+
+
+class ResourceGraphSource(BaseTest):
+
+    @arm_template('storage.json')
+    def test_resource_graph_and_arm_sources_storage_are_equivalent(self):
+        p1 = self.load_policy({
+            'name': 'test-azure-storage-arm-source',
+            'resource': 'azure.storage',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cctstorage*'}]
+        })
+
+        resources_arm = p1.run()[0]
+
+        p2 = self.load_policy({
+            'name': 'test-azure-storage-arm-source',
+            'resource': 'azure.storage',
+            'source': 'resource-graph',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cctstorage*'}]
+        }, validate=True)
+
+        resources_resource_graph = p2.run()[0]
+
+        self.assertTrue(resource_cmp(resources_arm, resources_resource_graph))
+
+    @arm_template('vm.json')
+    def test_resource_graph_and_arm_sources_vm_are_equivalent(self):
+        p1 = self.load_policy({
+            'name': 'test-azure-vm-arm-source',
+            'resource': 'azure.vm',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cctestvm*'}]
+        })
+
+        resources_arm = p1.run()[0]
+
+        p2 = self.load_policy({
+            'name': 'test-azure-vm-arm-source',
+            'resource': 'azure.vm',
+            'source': 'resource-graph',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cctestvm*'}]
+        }, validate=True)
```

Comment:
note we are trying to go back and ensure we only do validate true once, usually in a dedicated validate test method because of how slow it is (adds 2 seconds to each test).